### PR TITLE
#164798198 Documenting API with swagger

### PR DESCRIPTION
**Pull request by dorothyas**
**What does this PR do?**

- API documentation with swagger

**Description of Task to be completed?**

- Install django-rest-swagger
- Add url to the documentation in the main urls file
- Add rest_framework_swagger to installed apps in the settings file

**How should this be manually tested?**

-  Fetch the branch `git fetch origin ft-document-api-164798198`
-  Checkout the branch `git checkout ft-document-api-164798198`
-  Run `pipenv install`  to install the required packages
-  Run the command`python manage.py runserver`
-  http://127.0.0.1:8000/swagger-docs/

**What are the relevant pivotal tracker stories?**

- [#164798198](https://www.pivotaltracker.com/n/projects/2320796/stories/164798198)

**Screenshots**

Screenshot 2019-03-28 at 17 24 43
**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -10,16 +10,43 @@
...
-gunicorn
-jwt
-pipenv-to-requirements
-psycopg2-binary
-python-decouple
+asn1crypto==0.24.0
+certifi==2019.3.9
+cffi==1.12.2
+chardet==3.0.4
+coreapi==2.3.3
```
Comment (ja-odur) [inline requirements.txt]: @dorothyas Why is your `pipenv_to_requirements generating all these dependencies?

**Review by dannylwe — DISMISSED** [review]

File: authors/apps/authentication/views.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ def post(self, request):
         return Response(serializer.data, status=status.HTTP_200_OK)
 
 
-class UserRetrieveUpdateAPIView(RetrieveUpdateAPIView):
+class UserRetrieveUpdateAPIView(GenericAPIView):
```
Comment (ja-odur) [inline authors/apps/authentication/views.py]: @dorothyas I think this should be `GenericRetrieveUpdateAPIView` in order not to change the functionality of the endpoint.

**Review by ezrogha — DISMISSED** [review]
Great work

File: authors/apps/authentication/views.py (hunk 2 above)
Comment (dorothyas) [reply authors/apps/authentication/views.py]: okay

**Review by dannylwe — DISMISSED** [review]

**Review by felixkiryowa — DISMISSED** [review]

**Review by ezrogha — DISMISSED** [review]

**Review by ja-odur — APPROVED** [review]
LGTM

**Review by felixkiryowa — APPROVED** [review]
Working

**Review by dannylwe — APPROVED** [review]

**Pull request merged**